ngraph/src/ngraph/frontend/onnx_import/core/graph.cpp

```diff
@@ -61,73 +60,35 @@ namespace ngraph
                 std::string domain = get_node_domain(node_proto);
                 return (domain.empty() ? "" : domain + ".") + node_proto.op_type();
             }
-
-            static std::string concat_strings(
-                const std::vector<std::reference_wrapper<const std::string>>& strings)
-            {
-                const auto concat_with_comma =
-                    [](const std::string& accumulator,
-                       std::reference_wrapper<const std::string> next_string) {
-                        return accumulator + ", " + next_string.get();
-                    };
-
-                return std::accumulate(
-                    strings.begin() + 1, strings.end(), strings.begin()->get(), concat_with_comma);
-            }
-
-            static std::string build_input_provenance_tag(const std::string& input_name,
-                                                          const PartialShape& shape)
-            {
-                std::stringstream tag_builder;
-                tag_builder << "<ONNX Input (" << input_name << ") Shape:" << shape << ">";
-                return tag_builder.str();
-            }
-
-            static std::string build_op_provenance_tag(const Node& onnx_node)
-            {
-                const auto output_names = concat_strings(onnx_node.get_output_names());
-                const auto node_name =
-                    onnx_node.get_name().empty() ? "" : onnx_node.get_name() + " ";
-
-                return std::string{"<ONNX " + onnx_node.op_type() + " (" + node_name + "-> " +
-                                   output_names + ")>"};
-            }
         } // namespace detail
 
         Graph::Graph(const ONNX_NAMESPACE::GraphProto& graph_proto, Model& model)
+            : Graph(graph_proto, model, std::unique_ptr<GraphCache>(new GraphCache(graph_proto)))
```

Comment:
Please use `std::make_unique`

Reply:
I would like but it is unfortunately C++14 feature